core/src/main/java/org/infinispan/globalstate/impl/GlobalConfigurationManagerImpl.java

```diff
@@ -107,24 +109,32 @@ void start() {
          }
       }
 
+      // Install the listener
+      GlobalConfigurationStateListener stateCacheListener = new GlobalConfigurationStateListener(this);
+      getStateCache().addListener(stateCacheListener, new ScopeFilter(CACHE_SCOPE));
+
       // Initialize caches which are present in the initial state. We do this before installing the listener.
```

Comment:
This comment is no longer valid since you've moved things around

Reply:
True. I'm changing the listener to include the current state and I'm removing the explicit iteration.